README.adoc

```diff
@@ -212,19 +297,80 @@ IMPORTANT: This command need to be executed from the same directory where the in
 
 NOTE: See more over it in the Readme of https://github.com/matryer/moq[moq]
 
-== Developing Helpers
-=== Errcheck
+* Using the mock
 
-The https://github.com/kisielk/errcheck[errcheck] is required to run the `make errcheck` command.
+In the created file you will see an implementation commented as in the following example. This implementation will be used in the test file to mock the methods/func of this interface.
 
-Install:
+[source,go]
+----
+// AppServiceMock is a mock implementation of Service.
+//
+//     func TestSomethingThatUsesAppService(t *testing.T) {
+//
+//         // make and configure a mocked Service
+//         mockedAppService := &AppServiceMock{
+//             GetAppsFunc: func() (*[]models.App, error) {
+// 	               panic("mock out the GetApps method")
+//             },
+//         }
+//
+//         // use mockedAppService in code that requires Service
+//         // and then make assertions.
+//
+//     }
+----
 
-[source,shell]
+* Adding data to the mock function
+
+The `panic` statement needs to be replaced for the mock data. Following an example.
+
+[source,go]
+----
+numOfDeployedVersions := 5
+numOfAppLaunches := 1000
+numOfCurrentInstalls := 9000
+
+// mock data
+app := models.App{
+    ID:                    "a0874c82-2b7f-11e9-b210-d663bd873d93",
+    AppID:                 "com.aerogear.app1",
+    AppName:               "app1",
+    NumOfDeployedVersions: &numOfDeployedVersions,
+    NumOfAppLaunches:      &numOfAppLaunches,
+    NumOfCurrentInstalls:  &numOfCurrentInstalls,
+}
+
+// make and configure a mocked Service
+mockedAppService := &AppServiceMock{
+    GetAppsFunc: func() (*[]models.App, error) {
+        return &[]models.App{
+            app,
+        }, nil
+    },
+}
+----
+
+* Calling the mock
+
+You will call the mock instead of use the interface. It will return the data mocked as defined above. Following an practical example.
+
+[source,go]
 ----
-go get -u github.com/kisielk/errcheck
+got, err := mockedAppService.GetApps()
+if (err != nil) != tt.wantErr {
+    t.Errorf("appsService.GetApps() error = %v, wantErr %v", err, tt.wantErr)
+    return
+}
+if !reflect.DeepEqual(got, tt.want) {
+    t.Errorf("appsService.GetApps() = %v, want %v", got, tt.want)
+}
 ----
 
-=== Commands
+NOTE: The mock file contains explanations for example over how to check the quantity of results.
```

Comment:
@camilamacedo86 I'm not sure what you are trying to say here.
Is it? 
```suggestion
NOTE: The mock file contains explanations for how to check the quantity of results.
```

Reply:
@camilamacedo86 This one is still outstanding.

Reply:
Sure, you are 100% correct here. It is fixed now. Tks to get it.